src/main/java/org/opensearch/sdk/ActionListener.java

```diff
@@ -21,13 +21,35 @@
 import java.net.Socket;
 import java.net.UnknownHostException;
 
+/**
+ * A listener for actions on the local port.
+ */
 public class ActionListener {
 
+    /**
+     * Instantiate a new ActionListener.
```

Comment:
I am 100% with you on that preference.  Unfortunately the JDK maintainers decided in JDK 18 to require javadocs on constructors even if they are the default constructor!  I'm going to request this can get skipped in the `require-javadoc` plugin we're discussing in #27 so eventually this can get backed out.

Reply:
(See https://bugs.openjdk.org/browse/JDK-8249634)